test/e2e/storage/testsuites/provisioning.go

```diff
@@ -211,6 +211,11 @@ func (p *provisioningTestSuite) DefineTests(driver TestDriver, pattern testpatte
 		init()
 		defer cleanup()
 
+		if pattern.Name == testpatterns.BlockVolModeDynamicPV.Name || pattern.Name == testpatterns.DefaultFsDynamicPV.Name {
+			pattern.SnapshotType = testpatterns.DynamicCreatedSnapshot
```

Comment:
I think we should add the pattern to testpattern.go instead of dynamically changing the pattern.

We should also verify that the tests in snapshottable.go are also testing raw block.

Reply:
cc @chrishenzie who's also adding another snapshot test suite

Reply:
Also cc  @Jiawei0227

Reply:
Just checked snapshottable testsuite. We don't have raw block test coverage.
https://github.com/kubernetes/kubernetes/blob/3f6ec26eefcae1cdb4a43aa4e664a5f60fc7ca78/test/e2e/storage/testsuites/snapshottable.go#L76

And I feel like this testcase "should provision storage with snapshot data source" is already covered in the snapshottable test since that one is an e2e flow from create snapshot and restore from it, which also use dynamic provisioning.

Reply:
I think what happened here is this test case is a combination of provisioning ability + snapshot ability. So we either need to make a new testsuite to test this combination or just like what @qinpingli did here, modify it in-flight. 

Adding a new pattern and add it to the provisioning testsuite will just cause redundant test run for other testcases in this suite.

Reply:
I am actually leaning more toward removing this testcase from this testsuite because we have already test it in the snapshottable. And for snapshottable, we should also add rawblock coverage.

Reply:
Consolidating redundant tests sounds good to me.